### PR TITLE
Enable TensorFlow models on TPU multi-workers

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Enable TensorFlow models on TPU (with multi-workers):
* Enable to customize the TPU name, so that `--tpu=` config in Pod training can be known when passing run commands (I did not find a better way to obtain the tpu_name from XCom, as the value can only be retrieved within a task or python operator in Airflow).
* Write a few helper functions to construct model configs.
* Auto format the updated files.
* Other small changes, i.e. update args.

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** 
Upload to an Airflow environment, and trigger a run.

**List links for your tests (use go/shortn-gen for any internal link):** 
* Successfully triggered the pod test and got the same error as single host (SSL issue: [b/304394983](https://buganizer.corp.google.com/issues/304394983))
* Verified logs as expected, i.e. `Connecting to IP addresses of worker 0...` in [pod training](https://screenshot.googleplex.com/4szQQHw36WQaR2N), `export TPU_LOAD_LIBRARY=0 `[env variable](https://screenshot.googleplex.com/AkFMaEkHqLndtPN) & [custom tpu name](https://screenshot.googleplex.com/8CQLWRzbwuyJhgN).

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/xlml/solutionsTeam_flax_latest_supported.py [hunk 1]
```diff
@@ -67,4 +67,4 @@
   # Model dependencies
   jax_resnet_v4_8 >> jax_resnet_v4_32
   jax_gpt2_v4_8
-  jax_sd_v4_8
\ No newline at end of file
+  jax_sd_v4_8
```
Comment (will-cromar) [inline dags/xlml/solutionsTeam_flax_latest_supported.py:70]: Is this related to the rest of the PR?

**Review by will-cromar — APPROVED** [review]

File: dags/xlml/solutionsTeam_flax_latest_supported.py (hunk 1 above)
Comment (RissyRan) [reply dags/xlml/solutionsTeam_flax_latest_supported.py:70]: I think it adds a new line during the formatting.
